cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -389,26 +389,33 @@ externalSetupMethod verbosity options pkg bt mkargs = do
       Nothing    -> getInstalledPackages verbosity
                     comp (usePackageDB options') conf
 
-  cabalLibVersionToUse :: IO (Version, (Maybe ComponentId)
+  cabalLibVersionToUse :: IO (Version, Maybe ComponentId
                              ,SetupScriptOptions)
   cabalLibVersionToUse =
-    case useCabalSpecVersion options of
-      Just version -> do
+    case find (hasCabal . snd) (useDependencies options) of
```

Comment:
The code still needs to determine whether the solver chose the Cabal version, though, which would be hard to determine from `useCabalVersion`.  If the solver chose a Cabal version then we could use `useCabalVersion`.  But I didn't want to change the behavior in the case that the solver didn't choose any setup dependencies or chose setup dependencies that didn't include Cabal.